### ISSUE TITLE
Picking up items with , while inventory is full crashes.

### ISSUE THREAD

**Issue by Karcharoth** [NickMcConnell/NarSil#443]
Drat, I wrote this up and then accidentally lost it all. TL;DR: using ',' crashes the game if your inventory is full because it passes an object to player_pickup_item, unlike using 'g', and player_pickup_item tries to grab any item passed to it, inventory space or not. Meanwhile, the subfunction player_pickup_aux crashes if passed an item when there's no inventory space. Fixed [here](https://github.com/Karcharoth/Keys-of-Orthanc/commit/3a695c1dc736cf3a84374c478e5ca3882cf9be53).

**Comment by backwardsEric** [comment]
This is also a potential problem in Vanilla Angband:  all current uses of player_pickup_item() there seem to check before passing an item to it but the function's documentation does not specify whether it tolerates cases where the item cannot  be picked up.  Since it does tolerate full inventories when obj is NULL, Karcharoth's fix seems right, but I'd not fall through and do the rest of the function if obj is non-NULL and the inventory is full so something like this for NarSil:

```
    if (obj) {
        mem_free(floor_list);
        if (inven_carry_num(p, obj) > 0) {
            player_pickup_aux(p, obj, 0, domsg);
        }
        return;
    }
```

The equivalent change for Vanilla Angband is up as https://github.com/angband/angband/pull/5983 .

**Comment by NickMcConnell** [comment]
Thanks, done by cherry-picking the V fix and adjusting in 1bf26cb.
